src/api.d.ts

```diff
@@ -12,81 +35,168 @@ export interface TreeNodeConfiguration {
+     */
+    deleteTreeItemImpl?(context: IActionContext): Promise<void>;
+
+    /**
+     * Implement this to execute any async code when this node is refreshed. Should not be called directly
+     */
+    refreshImpl?(context: IActionContext): Promise<void>;
+
+    /**
+     * Optional function to filter items displayed in the tree picker. Should not be called directly
+     * If not implemented, it's assumed that 'isAncestorOf' evaluates to true
+     */
+    isAncestorOfImpl?(contextValue: string | RegExp): boolean;
 }
 
-export interface GroupableResource {
-    readonly groupConfig: GroupingConfig;
-}
+export type ResolvedAppResourceTreeItemBase = Partial<{ [P in keyof SealedAzExtTreeItem]: never }> & AbstractAzExtTreeItem;
 
-export interface GroupableApplicationResource extends ApplicationResource {
-    readonly groupConfig: GroupingConfig;
-}
+export type ResolvedAppResource = ResolvedAppResourceTreeItemBase;
+
+export type ResolvedAppResourceTreeItem<T extends ResolvedAppResourceTreeItemBase> = AppResource & Omit<T, keyof ResolvedAppResourceTreeItemBase>;
 
 export type LocalResource = AzExtTreeItem;
 
-export interface ApplicationResourceProvider {
-    provideResources(subContext: ISubscriptionContext): vscode.ProviderResult<GenericResource[] | undefined>;
+export interface AppResourceResolver {
+    resolveResource(subContext: ISubscriptionContext, resource: AppResource): vscode.ProviderResult<ResolvedAppResource>;
 }
 
-export interface ApplicationResourceResolver {
-    resolveResource(subContext: ISubscriptionContext, resource: GenericResource): vscode.ProviderResult<ResolveResult>;
+/**
+ * Resource extensions call this to register app resource resolvers.
+ *
+ * @param resolver
+ * @param resourceType
+ * @param resourceKind
+ */
+export declare function registerApplicationResourceResolver(resolver: AppResourceResolver, resourceType: string, resourceKind?: string): vscode.Disposable;
+
+// Not part of public interface to start with--only Resource Groups extension will call it (for now)
+// currently implemented as AzureResourceProvider
+export interface AppResourceProvider {
+    provideResources(
+        subContext: ISubscriptionContext
+    ): vscode.ProviderResult<AppResource[]>;
 }
 
 export interface LocalResourceProvider {
     provideResources(): vscode.ProviderResult<LocalResource[] | undefined>;
 }
 
-// called from a resource extension (SWA, Functions, etc)
-export declare function registerApplicationResourceResolver(
-    provider: ApplicationResourceResolver,
-    resourceType: string,
-    resourceKind?: string,
-): vscode.Disposable;
-
 // Resource Groups can have a default resolve() method that it supplies, that will activate the appropriate extension and give it a chance to replace the resolve() method
 // ALSO, it will eliminate that default resolver from future calls for that resource type
 
 // called from host extension (Resource Groups)
```

Comment:
```suggestion
```
Not the case anymore

Reply:
`registerApplicationResourceProvider` is still called from the host extension right?

Reply:
You're right, my brain was thinking this was the register...Resolver which is called from the client extensions. The host extension will call `registerApplicationResourceProvider` (on itself).